### PR TITLE
mdk4: init at unstable-2021-04-27

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
#81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/mdk3/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  buildInputs = [ libnl libpcap ];
+
+  installFlags = [ "PREFIX=$(out)" "SBINDIR=$(PREFIX)/bin" ];
+
+  meta = with lib; {
+    description = "A tool that injects data into wireless networks";
+    homepage = "https://github.com/aircrack-ng/mdk3";
+    maintainers = with maintainers; [ fortuneteller2k ];
+    license = licenses.gpl3Only;
```
Comment (fabaff) [inline pkgs/tools/networking/mdk3/default.nix:24]: Looks like [GPLv2+](https://github.com/aircrack-ng/mdk3/blob/master/src/osdep/dummy.c#L6-L9).

```suggestion
    license = licenses.gpl2Plus;
```

**Review by fabaff — COMMENTED** [review]
My concern here is that the repo was archived and that there was no activity in the last 5 years. Thus, bugs will not be fixed by the upstream project.

What about `mdk4`? Looks like the successor.

The repo contains a [man](https://github.com/aircrack-ng/mdk3/blob/master/man/mdk3.8) page.

**Comment by SuperSandro2000** [comment]
mdk3 repo is archived. Maybe we only need mdk4?

**Comment by fortuneteller2k** [comment]
Moved to mdk4, had to use the latest commit on master since the Makefile of the released version was giving me some trouble.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 124354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mdk4</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set

- [x] package built (see https://github.com/NixOS/nixpkgs/pull/124354#issuecomment-848522793)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-124354]$ ./results/mdk4/bin/mdk4 
MDK4 4.2 - "Awesome! Supports Proof-of-concept of WiFi protocol implementation vulnerability testing"
by E7mer, thanks to the aircrack-ng community.
MDK4 is a proof-of-concept tool to exploit common IEEE 802.11 protocol weaknesses.
IMPORTANT: It is your responsibility to make sure you have permission from the
network owner before running MDK4 against it.

This code is licenced under the GPLv2 or later

MDK4 USAGE:
mdk4 <interface> <attack_mode> [attack_options]
[...]           
```

Looks good to me.

**Pull request merged**